pkg/commands/resolver.go

```diff
@@ -470,3 +472,19 @@ func resolveFile(
 
 	return buf.Bytes(), nil
 }
+
+// create a set from the input slice
+// preserving the order of unique elements
+func unique[T comparable](tt []T) []T {
```

Comment:
Ehh, I don't know if we need to bother with generics here, since we know we'll only call it with strings.

Reply:
Yes, I had it only for strings before, but then I thought it may be useful for other stuff too. You mentioned platform, for example. Maybe that's also a string, not sure. I can change it, if you want, though.

Reply:
Yeah I'd prefer to have it work on strings until/unless we find a generic use for it in the future. YAGNI and all.

Reply:
ok done. I did a force push. Not sure how this is done here, but I know some project not doing squash merge and thefore prefer a single commit.